backend/db.ts

```diff
@@ -1,66 +1,83 @@
-const _ = require('lodash');
-const fs = require('fs');
-const request = require('request');
-const Sequelize = require('sequelize');
+import _ from 'lodash';
+import fs from 'fs';
+import request from 'request';
+import { DataTypes, Model, ModelCtor, Options, Sequelize, QueryTypes, QueryOptionsWithType } from 'sequelize';
+
+import type { LoggerFactory } from './logger';
+
+type Db = { sequelize?: Sequelize } & Record<string, Model>;
+type RestartFunction = (reason: string) => void;
+export type DbModule = { db: Db; init: () => Promise<void> };
 
 /**
  * Once initialized this object contains `sequelize` instance as well as all DB models, keyed by model name.
  */
-const db = {};
+const db: Db = {};
 
-function wait(seconds) {
+function wait(seconds: number) {
     return new Promise(resolve => setTimeout(resolve, 1000 * seconds));
 }
 
-function addHooks(sequelize, restart) {
-    sequelize.afterDisconnect(async ({ _invalid: unexpectedDisconnection }) => {
+function addHooks(sequelize: Sequelize, restart: RestartFunction) {
+    // @ts-ignore afterDisconnect is static method
+    sequelize.afterDisconnect(async (connection: unknown) => {
+        const { _invalid: unexpectedDisconnection } = connection as { _invalid: boolean };
         if (unexpectedDisconnection) {
             restart('Unexpected disconnection occured.');
         }
     });
 
-    sequelize.beforeQuery(async ({ isRecoveryCheck }) => {
+    // @ts-ignore beforeQuery is not included in TypeScript declaration for sequelize,
+    // See: https://github.com/sequelize/sequelize/issues/11441
+    sequelize.beforeQuery(async ({ isRecoveryCheck }: { isRecoveryCheck: boolean }) => {
+        // eslint-disable-next-line camelcase
+        type PgIsInRecoveryQueryResponse = { pg_is_in_recovery: boolean };
         if (!isRecoveryCheck) {
-            const result = await sequelize.query('SELECT pg_is_in_recovery();', {
+            const result = (await sequelize.query('SELECT pg_is_in_recovery();', {
                 plain: true,
-                type: Sequelize.QueryTypes.SELECT,
+                type: QueryTypes.SELECT,
                 isRecoveryCheck: true
-            });
+            } as QueryOptionsWithType<QueryTypes.SELECT> & { plain: true })) as PgIsInRecoveryQueryResponse;
             if (result && result.pg_is_in_recovery) {
                 restart('DB is in recovery.');
             }
         }
     });
 }
 
+export type DbConfig = { url: string | string[]; options: Pick<Options, 'dialectOptions'> };
+export type ModelFactories = ((sequelize: Sequelize, dataTypes: typeof DataTypes) => ModelCtor<any>)[];
```

Comment:
`sequelize.define` returns ModelCtor. I'm not sure yet how to properly define models in consumers (stage / composer) to provide proper type and not use `any` here.